mlrun/utils/notifications/notification/mail.py

```diff
@@ -0,0 +1,153 @@
+
+    required_params = [
+        "server_host",
+        "server_port",
+        "sender_address",
+        "username",
+        "password",
+        "email_addresses",
+    ] + boolean_params
+
+    @classmethod
+    def validate_params(cls, params):
+        for required_param in cls.required_params:
+            if required_param not in params:
+                raise ValueError(
+                    f"Parameter '{required_param}' is required for MailNotification"
+                )
+
+        for boolean_param in cls.boolean_params:
+            if not isinstance(params.get(boolean_param, None), bool):
+                raise ValueError(
+                    f"Parameter '{boolean_param}' must be a boolean for MailNotification"
+                )
+
+        cls._validate_email_addresses(params)
+
+    async def push(
+        self,
+        message: str,
+        severity: typing.Optional[
+            typing.Union[mlrun.common.schemas.NotificationSeverity, str]
+        ] = mlrun.common.schemas.NotificationSeverity.INFO,
+        runs: typing.Optional[typing.Union[mlrun.lists.RunList, list]] = None,
+        custom_html: typing.Optional[typing.Optional[str]] = None,
+        alert: typing.Optional[mlrun.common.schemas.AlertConfig] = None,
+        event_data: typing.Optional[mlrun.common.schemas.Event] = None,
+    ):
+        self.params.setdefault("subject", f"[{severity}] {message}")
+        self.params.setdefault("body", message)
+        await self._send_email(**self.params)
+
+    @classmethod
+    def enrich_default_params(
+        cls, params: dict, default_params: typing.Optional[dict] = None
+    ) -> dict:
+        params = super().enrich_default_params(params, default_params)
+        params.setdefault("use_tls", True)
+        params.setdefault("start_tls", False)
+        params.setdefault("validate_certs", True)
+        params.setdefault("server_port", DEFAULT_SMTP_PORT)
+
+        default_mail_address = params.pop("default_email_addresses", "")
+        email_addresses = params.get("email_addresses", default_mail_address)
+        if isinstance(email_addresses, list):
+            email_addresses = ",".join(email_addresses)
+        params["email_addresses"] = email_addresses
+
+        return params
+
+    @classmethod
+    def _validate_email_addresses(cls, params):
+        sender_address = params["sender_address"]
+        if not re.match(mlrun.utils.regex.mail_regex, sender_address):
+            raise ValueError(
+                f"Invalid email address '{sender_address}' in 'sender_address'"
+            )
+
+        if not isinstance(params["email_addresses"], (str, list)):
+            raise ValueError(
+                "Parameter 'email_addresses' must be a string or a list of strings"
+            )
+
+        email_addresses = params["email_addresses"]
+        if isinstance(params["email_addresses"], str):
+            email_addresses = params["email_addresses"].split(",")
```

Comment:
maybe create a small method that does this to eliminate the code duplication in the file